### PR TITLE
Switch ncs to use radio 1.6

### PR DESCRIPTION
**Pull request by MarekPorwisz**
Moving back to radio 1.6
Samples updated to work with this radio version
Docs updated.

### CODE REVIEW

**Comment by wbober** [comment]
LGTM. 

Does it work?

File: west.yml [hunk 1]
```diff
@@ -46,7 +48,8 @@ manifest:
     # https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/modules.html
     - name: zephyr
       repo-path: sdk-zephyr
-      revision: ad0444b058ef5b3586784425c5a353cd5fe9a315
+      revision: pr/remove-nrf-802154-sl-choice
```
Comment (hubertmis) [inline west.yml:51]: Shall we change it to pull/318/head ?

**Comment by rlubos** [comment]
What's with the merge commit?

**Comment by hubertmis** [comment]
Shall we squash it? I don't think that most of the commits in this PR work without complete changes.

File: west.yml [hunk 2]
```diff
@@ -96,11 +100,7 @@ manifest:
     - name: nrfxlib
       repo-path: sdk-nrfxlib
       path: nrfxlib
-      revision: 705d00a71149b876abbbe681631be0b89de83334
-    - name: hal_nordic
-      repo-path: sdk-hal_nordic
-      path: modules/hal/nordic
-      revision: ef6c7c3e82fee6e9e58c5295a4de7201b4b99b3c
+      revision: radio_16_support
```
Comment (jciupis) [inline west.yml:103]:
```suggestion
      revision: pull/203/head
```

https://github.com/nrfconnect/sdk-nrfxlib/pull/203

Comment (maciekfabia) [inline west.yml]: we can replace this one with `pull/203/head`

Comment (jciupis) [reply west.yml]: https://github.com/nrfconnect/sdk-nrf/commit/829684b16a22e7a3fbbf9f0a060853c2ccb0b33a

Done already

**Comment by rlubos** [comment]
Dependencies merged, please update SHA.

**Pull request merged**